core/row/row.selector.js

```diff
@@ -0,0 +1,145 @@
+import {AppError} from '../infrastructure';
+import {getFactory, get} from '../services/label';
+import {isUndefined} from '../utility/utility';
+
+export class RowSelector {
+	constructor(model) {
+		this.model = model;
+	}
+
+	map(items) {
+		const selectionState = this.model.selection();
+
+		switch (selectionState.unit) {
+			case 'row':
+				return this.mapFromRows(items);
+			case 'column':
+				return this.mapFromColumns(items);
+			case 'cell':
+				return this.mapFromCells(items);
+			case 'mix':
+				return this.mapFromMix(items);
+			default:
+				throw new AppError('row.selector', `Invalid unit ${selectionState.unit}`);
+		}
+	}
+
+	mapFromRows(rows) {
+		const result = [];
+		const cache = new Map();
+		const columns = this.model
+			.view()
+			.columns
+			.filter(column => column.class === 'data' || column.class === 'pivot');
+
+		for (const row of rows) {
+			const line = [];
+
+			for (const column of columns) {
+				let label;
+				const key = column.key;
+
+				if(cache.has(key)) {
+					label = cache.get(key)
+				} else {
+					label = getFactory(column);
+					cache.set(key, label);
+				}
+
+				const value = label(row);
+				line.push(value === null || isUndefined(value) ? '' : '' + value);
+			}
+
+			result.push(line);
+		}
+
+		return result;
+	}
+
+	mapFromColumns(columns) {
+		const result = [];
+		const rows = this.model.view().rows;
+		const cache = new Map();
+
+		for(let i = 0, columnLength = columns.length; i < columnLength; i++) {
```

Comment:
make a method `mapFromRowColumns(rows, columns)`
and use there `mapFromRows` body
inside `mapFromRows` and `mapFromColumns` use mapFromRowColumns